src/Compilers/Core/MSBuildTask/MvidReader.cs

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.BuildTasks
+{
+    public sealed class MvidReader : BinaryReader
```

Comment:
I'll make the constructor private.
I gave your suggestion a shot (making the class static, but it made the code worse). For instance, I need to add `ReadUInt32` and `ReadUInt16` wrapper methods, and then pass a `BinaryReader` to pretty much every line in this class.